dlc-messages/src/contract_msgs.rs

```diff
@@ -178,9 +178,11 @@ pub struct PayoutFunction {
     pub payout_function_pieces: Vec<PayoutFunctionPiece>,
     /// The right most point of the function.
     pub last_endpoint: PayoutPoint,
+    /// An optional fee deducted by the offering party in the payouts.
+    pub fee: Option<u64>,
```

Comment:
Hmm, this is not what I had in mind. In fact I thought we would be able to do this without changing `rust-dlc` at all.

After experimenting a few weeks ago I concluded that you can apply the fee offset to the input (i.e. the margins) and it has the same effect as applying it on the output (i.e. the payouts).